components/style/values/generics/grid.rs

```diff
@@ -475,6 +457,16 @@ impl<L: Clone> TrackRepeat<L> {
     }
 }
 
+/// Track list values. Can be <track-size> or <track-repeat>
+#[derive(Clone, Debug, PartialEq, ToComputedValue, ToCss)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+pub enum TrackListValue<T> {
```

Comment:
TBH I would prefer to keep the current enum if you don't mind. Either<T> syntax feels less clear than an explicit enum, specially since we need to do things like https://github.com/servo/servo/pull/18206/files#diff-ea526b2e68fb17c4d9af5ce82d3cc64eR899